File: Data Structures & Algorithms/is-palindrome/submission-3.java
class Solution {
    public boolean isPalindrome(String s) {
       s=s.toLowerCase(Locale.ROOT);
       int left =0,right=s.length()-1;
       while(left<=right)
       {
           while (left<s.length() && !Character.isLetterOrDigit(s.charAt(left)))
           {
               left++;
           }
           while   (right>=0 && !Character.isLetterOrDigit(s.charAt(right)))
           {
               right--;
           }
           if(left<s.length() && right>=0 && s.charAt(right)!=s.charAt(left))
           {
               return false;
           }

           left++;right--;
       }
        return true; 
    }
}

File: Data Structures & Algorithms/binary-tree-from-preorder-and-inorder-traversal/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def buildTree(self, preorder: List[int], inorder: List[int]) -> Optional[TreeNode]:
        def dfs(preorder,inorder):
            if not (preorder or inorder):
                return None
            root = TreeNode()
            rvalue = preorder[0]
            root.val = rvalue
            rindex = inorder.index(rvalue)
            root.left = dfs(preorder[1:rindex+1],inorder[:rindex])
            root.right = dfs(preorder[rindex+1:],inorder[rindex+1:])

            return root
        return dfs(preorder,inorder)